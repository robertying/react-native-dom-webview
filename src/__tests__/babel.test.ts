import { transformAsync } from '@babel/core';
const plugin = require('../babel');

const input = `
import React from "react";
import { View, Text } from "react-native";

const a = "123";

const app = DOM("./web/App");
const editor = DOM("./Editor");

const fileList = preval.require("./lib/files", 3);

export default function MyScreen() {
    return (
        <View>
            <Text>abc</Text>
        </View>
    )
}
`;

it('replaces `DOM` with `preval.require`', async () => {
  const result = await transformAsync(input, {
    babelrc: false,
    filename: 'MyScreen.tsx',
    plugins: [
      [
        plugin,
        {
          packagerPort: 8081,
        },
      ],
    ],
  });
  expect(result?.code).toMatchSnapshot();
});
