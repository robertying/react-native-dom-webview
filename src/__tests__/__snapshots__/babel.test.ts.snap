// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`replaces \`DOM\` with \`preval.require\` 1`] = `"var _interopRequireDefault=require(\\"@babel/runtime/helpers/interopRequireDefault\\");Object.defineProperty(exports,\\"__esModule\\",{value:true});exports.default=MyScreen;var _react=_interopRequireDefault(require(\\"react\\"));var _reactNative=require(\\"react-native\\");var a=\\"123\\";var app=preval.require(\\"<PROJECT_ROOT>/client\\",\\"8081\\",\\"web/App\\");var editor=preval.require(\\"<PROJECT_ROOT>/client\\",\\"8081\\",\\"Editor\\");var fileList=preval.require(\\"./lib/files\\",3);function MyScreen(){return _react.default.createElement(_reactNative.View,null,_react.default.createElement(_reactNative.Text,null,\\"abc\\"));}"`;
